koji/playbooks/make-modules-metadata.yaml

```diff
@@ -0,0 +1,31 @@
+---
+- hosts: localhost
+  gather_facts: no
+  tasks:
+    - name: Collect RPM names
+      # The epoch number in the NEVRA string is mandatory per spec
+      # https://github.com/fedora-modularity/libmodulemd/blob/main/yaml_specs/modulemd_stream_v2.yaml#L668
+      # So we can't just use `%{nevra}` here
+      command: "rpm --query --package {{ repository_path }}/*rpm --queryformat='%{name}-%{epochnum}:%{version}-%{release}.%{arch}\n'"
+      register: rpm_names
+
+    - name: Split RPM names into list
+      set_fact:
+        rpm_list: "{{ rpm_names.stdout_lines }}"
+
+    - name: Read base modules file
+      set_fact:
+        modules: "{{ lookup('file', modules_base) | from_yaml }}"
+
+    - name: Get current date for metadata version
+      set_fact:
+        version: "{{ lookup('pipe', 'date --universal +%Y%m%d%H%M%S') }}"
```

Comment:
hmm, I think I want to prepend this with the Foreman version, as otherwise a later generated metadata of Foreman 3.0 might look "newer" than the one from 3.1.

(I have no idea what this "version" is used for in DNF, it seems to work just fine with 0, but the spec says it needs to be set, and an unsigned 64bit int can easily fit that information)

Reply:
So would you do something like `(major * 100 + minor) * 10 ^ 14 + date`?

Reply:
something like that, yeah